### PR TITLE
Add validation to network.vde[*].name fields

### PR DESCRIPTION
**Pull request by jandubois**
Not really happy about the `qemuconst` package, but `limayaml` importing `qemu` would create a circular dependency.

Signed-off-by: Jan Dubois <jan.dubois@suse.com>

### CODE REVIEW

File: pkg/qemu/qemuconst/qemuconst.go [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package qemuconst
+
+const (
+	SlirpMACAddress  = "22:11:11:11:11:11"
+	SlirpNetworkName = "eth0"
```
Comment (AkihiroSuda) [inline pkg/qemu/qemuconst/qemuconst.go:5]: I'd call this "SlirpNICName"

File: pkg/qemu/qemuconst/qemuconst.go [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package qemuconst
+
+const (
+	SlirpMACAddress  = "22:11:11:11:11:11"
```
Comment (AkihiroSuda) [inline pkg/qemu/qemuconst/qemuconst.go:4]: Maybe we should hash SlirpMACAddress too, but can be another PR.

File: pkg/qemu/qemuconst/qemuconst.go (hunk 1 above)
Comment (jandubois) [reply pkg/qemu/qemuconst/qemuconst.go]: changed

File: pkg/qemu/qemuconst/qemuconst.go (hunk 2 above)
Comment (jandubois) [reply pkg/qemu/qemuconst/qemuconst.go:4]: We could, but does it matter? Isn't it always on a network by itself?

**Review by AkihiroSuda — APPROVED** [review]
Thanks

File: pkg/qemu/qemuconst/qemuconst.go (hunk 2 above)
Comment (AkihiroSuda) [reply pkg/qemu/qemuconst/qemuconst.go:4]: Some apps may want eth0 MAC to be unique.

Comment (jandubois) [reply pkg/qemu/qemuconst/qemuconst.go:4]: Ok, I'll add it to this PR.

Is the name `eth0` fine, or should it be something like `slirp` instead?

I think I prefer to always have an `eth0` interface though; it feels more traditional. 😄

Comment (AkihiroSuda) [reply pkg/qemu/qemuconst/qemuconst.go:4]: eth0 looks good to me

Comment (jandubois) [reply pkg/qemu/qemuconst/qemuconst.go:4]: And any suggestion about where to put `qemuconst.SlirpNICName`? It feels a little silly to have a separate package just for one constant.

Comment (AkihiroSuda) [reply pkg/qemu/qemuconst/qemuconst.go:4]: >  It feels a little silly to have a separate package just for one constant.

I think this is fine

Comment (jandubois) [reply pkg/qemu/qemuconst/qemuconst.go]: > Maybe we should hash SlirpMACAddress too

Implemented in 5f4d187.

I'm slightly worried that the SlirpMACAddress is calculated in 2 different places, and that if only one of them is updated later, the addresses would no longer match. Not sure how to avoid this though.
